### PR TITLE
CLOUDP-99402: Live Migration Link-Token e2e

### PR DESCRIPTION
**Pull request by colm-quinn**
## Proposed changes
Adding E2E test to verify link-token functionality works correctly.

_Jira ticket:_ https://jira.mongodb.org/browse/CLOUDP-99402

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

Example run with an existing link-token (cleanup removes it):
```
=== RUN   TestLinkToken
2021/09/02 15:51:25 Removing link-tokens from previous/parallel test runs...
2021/09/02 15:51:26 Warning: Deleted link-token.
2021/09/02 15:51:26 Cleanup complete.
=== RUN   TestLinkToken/Create
=== RUN   TestLinkToken/Delete
--- PASS: TestLinkToken (1.77s)
    --- PASS: TestLinkToken/Create (0.53s)
    --- PASS: TestLinkToken/Delete (0.52s)
PASS
ok  	github.com/mongodb/mongocli/e2e/atlas	1.885s
```

Normal run:
```
=== RUN   TestLinkToken
2021/09/02 15:51:33 Removing link-tokens from previous/parallel test runs...
2021/09/02 15:51:34 Cleanup complete.
=== RUN   TestLinkToken/Create
=== RUN   TestLinkToken/Delete
--- PASS: TestLinkToken (1.80s)
    --- PASS: TestLinkToken/Create (0.61s)
    --- PASS: TestLinkToken/Delete (0.53s)
PASS
ok  	github.com/mongodb/mongocli/e2e/atlas	1.927s
```

### CODE REVIEW

File: e2e/E2E-TESTS.md [hunk 1]
```diff
@@ -322,8 +322,10 @@ Command                                         | E2E Atlas | E2E OM    | E2E CM
 | `performanceAdvisor namespace list`           |           | N         | N         |           | Y     | Y     |
 | `performanceAdvisor slowQueryLogs list`       |           | N         | N         |           | Y     | Y     |
 | `performanceAdvisor suggestedIndexes list`    |           | N         | N         |           | Y     | Y     |
-| `serverless create`                    | Y         |           |           | Y         |       |       |
```
Comment (colm-quinn) [inline e2e/E2E-TESTS.md:325]: Did an alignment cleanup while here

File: internal/cli/atlas/livemigrations/link/delete.go [hunk 2]
```diff
@@ -61,6 +61,7 @@ func DeleteBuilder() *cobra.Command {
 			return opts.Run()
 		},
 	}
+	cmd.Flags().BoolVar(&opts.Confirm, flag.Force, false, usage.Force)
```
Comment (colm-quinn) [inline internal/cli/atlas/livemigrations/link/delete.go:64]: Adding so we can disable the confirm for the e2e tests. Spec updated

File: e2e/atlas/live_migrations_test.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 MongoDB Inc
```
Comment (gssbzn) [inline e2e/atlas/live_migrations_test.go:1]:
```suggestion
// Copyright 2021 MongoDB Inc
```

File: e2e/atlas/live_migrations_test.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+func TestLinkToken(t *testing.T) {
+	cliPath, err := e2e.Bin()
+	r := require.New(t)
+	r.NoError(err)
+
+	// Cleanup, do a delete in case a token already exists from another run of the test
+	log.Printf("Removing link-tokens from previous/parallel test runs...")
```
Comment (gssbzn) [inline e2e/atlas/live_migrations_test.go:35]: [nit] you can use the test log https://pkg.go.dev/testing#T.Log
```suggestion
	t.Logf("Removing link-tokens from previous/parallel test runs...\n")
```

File: internal/cli/atlas/livemigrations/link/delete_test.go [hunk 5]
```diff
@@ -20,9 +20,12 @@ package link
 import (
 	"testing"
 
+	"github.com/mongodb/mongocli/internal/flag"
+
```
Comment (gssbzn) [inline internal/cli/atlas/livemigrations/link/delete_test.go:24]: this is a bit of an edge-case for goimport it doesn't sort after the first group 🤷 
```suggestion
	"github.com/mongodb/mongocli/internal/flag"
```

**Review by gssbzn — COMMENTED** [review]
main change to use the test logging other than that great stuff, the more tests the better

File: e2e/atlas/live_migrations_test.go (hunk 3 above)
Comment (colm-quinn) [reply e2e/atlas/live_migrations_test.go:1]: G*D D**N it, I keep checking those and missing one when doing a copy/pasta :D

File: internal/cli/atlas/livemigrations/link/delete_test.go (hunk 5 above)
Comment (colm-quinn) [reply internal/cli/atlas/livemigrations/link/delete_test.go:24]: Huh, good to know! (I was trusting IntelliJ to do this)

File: e2e/atlas/live_migrations_test.go (hunk 4 above)
Comment (colm-quinn) [reply e2e/atlas/live_migrations_test.go:35]: Thanks, hunted around for a info/warn/error log handler first, didn't think of Log 🤦

**Review by gssbzn — APPROVED** [review]
LGTM thanks for the changes
